backends/dpdk/dpdkArch.cpp

```diff
@@ -1624,16 +1624,16 @@ const IR::Node* CopyMatchKeysToSingleStruct::preorder(IR::Key* keys) {
     } else {
         structure->table_type_map.emplace(table->name.name, InternalTableType::WILDCARD);
     }
+
     /* If copyNeeded is false at this point, it means the keys are from same struct.
```

Comment:
why it was not working earlier?

Reply:
the "metaCopyNeeded" variable was not being set if the "copyNeeded" was already true due to keys coming from different struct. Learner and exact match table must have keys which are consecutive in header/metadata struct irrespective of whether the keys are from same struct or different structs, i.e. irrespective of whether copyNeeded is true or false.